onmt/decoders/transformer.py

```diff
@@ -45,22 +53,36 @@ def __init__(self, size, dropout,
         self.register_buffer('mask', mask)
 
     def forward(self, inputs, memory_bank, src_pad_mask, tgt_pad_mask,
-                previous_input=None):
-        # Args Checks
-        input_batch, input_len, _ = inputs.size()
-        if previous_input is not None:
-            pi_batch, _, _ = previous_input.size()
-            aeq(pi_batch, input_batch)
-        contxt_batch, contxt_len, _ = memory_bank.size()
-        aeq(input_batch, contxt_batch)
+                previous_input=None, layer_cache=None, step=None):
+        """
+        Args:
```

Comment:
Please make these comments in the same style as the rest of the code, as it is autodoc'd. Please see other places in the code. Particularly the return format.